test/linalg.jl

```diff
@@ -53,6 +53,43 @@ using Test
             end
         end
 
+        @testset "eigen decomposition" begin
+            # this hits both eigen(BlockDiagonal) and blockEigen, since the former uses the latter
+            evalsBlock, evecsBlock = eigen(b1)
+            evals, evecs = eigen(Matrix(b1))
+            @static if VERSION < v"1.2"
+                # pre-v1.2 we need to sort the eigenvalus consistently
+                sortPermBlock = sortperm(abs2.(evalsBlock))
+                evalsBlock = evalsBlock[sortPermBlock]
+                evecsBlock = evecsBlock[:, sortPermBlock]
+                sortPerm = sortperm(abs2.(evals))
+                evals = evals[sortPerm]
+                evecs = evecs[:, sortPerm]
+            end
+            @test evalsBlock ≈ evals
+            # comparing vectors is more difficult due to a sign ambiguity
+            # So we try adding and subtracting the vectors, keeping the smallest magnitude for each entry
+            # and compare that with something small.
+            # I performed some tests and the largest deviation I found was ~5e-14
+            @test all(min.(abs.(evecsBlock - evecs), abs.(evecsBlock + evecs)) .< 1e-13)
+
+            # also try for Symmetric matrices
```

Comment:
Could you make this, and the previous one separate `@testset`s? And add another `@testset` unit testing the `eigen_blockwise` function?